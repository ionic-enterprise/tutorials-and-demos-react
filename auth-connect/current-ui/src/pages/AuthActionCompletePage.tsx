import { useEffect } from 'react';
import { Capacitor } from '@capacitor/core';
import { IonContent, IonPage, IonSpinner } from '@ionic/react';
import { useHistory } from 'react-router-dom';
import { handleAuthCallback } from '../utils/authentication';
import './AuthActionCompletePage.css';

const AuthActionCompletePage: React.FC = () => {
  const history = useHistory();

  useEffect(() => {
    if (!Capacitor.isNativePlatform()) {
      handleAuthCallback().then(() => history.replace('/'));
    }
  }, [handleAuthCallback]);

  return (
    <IonPage>
      <IonContent className="main-content auth-action-complete">
        <div className="container">
          <IonSpinner name="dots" />
        </div>
      </IonContent>
    </IonPage>
  );
};
export default AuthActionCompletePage;
