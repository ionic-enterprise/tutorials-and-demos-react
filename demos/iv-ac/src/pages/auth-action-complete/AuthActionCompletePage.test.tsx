import { render } from '@testing-library/react';
import AuthActionCompletePage from './AuthActionCompletePage';

describe('<AuthActionCompletePage />', () => {
  it('renders consistently', () => {
    const { asFragment } = render(<AuthActionCompletePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
