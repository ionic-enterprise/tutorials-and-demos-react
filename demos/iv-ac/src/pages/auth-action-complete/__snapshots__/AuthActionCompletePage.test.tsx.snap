// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AuthActionCompletePage /> > renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-content
      class="main-content auth-action-complete"
    >
      <div
        class="container"
      >
        <ion-spinner
          name="dots"
        />
      </div>
    </ion-content>
  </div>
</DocumentFragment>
`;
